src/containers/RegistrationContainer.js

```diff
@@ -28,43 +36,116 @@ const RegistrationContainer = props => {
     mattermostId,
     interestOptions,
     registrationError,
+    getMe,
+    userBirthdate,
   } = props
   const [nickname, setNickname] = useState('')
-  const [showAge, setShowAge] = useState(false)
+  const [showAge, setShowAge] = useState('')
   const [location, setLocation] = useState('')
+  const [showLocation, setShowLocation] = useState('')
   const [description, setDescription] = useState('')
   const [img, setImg] = useState(null)
   const [interests, setInterests] = useState([])
+  const [nextButtonActive, setNextButtonActive] = useState(true)
+
+  useEffect(() => {
+    getMe()
+    props.addUserToStateAction()
+  }, [])
+
+  // Change nextButtonActive value only if new value is different
+  const setNextButtonStatus = value => {
+    if (value === true && !nextButtonActive) {
+      setNextButtonActive(true)
+    } else if (value === false && nextButtonActive) {
+      setNextButtonActive(false)
+    }
+  }
+
+  const checkInputValidity = page => {
```

Comment:
Just a nitpick, but this might be even cleaner as a switch-case like subpage switch :)

Reply:
Agreed, changed the if chain to switch-case :)